src/main/java/org/codehaus/groovy/ast/ClassNode.java

```diff
@@ -1420,6 +1421,16 @@ public void setUsingGenerics(boolean usesGenerics) {
         this.usesGenerics = usesGenerics;
     }
 
+    public ClassNode getPlainPrimitiveReference() {
+        ClassNode n = new ClassNode(name, modifiers, superClass, null, null);
+        n.isPrimaryNode = false;
+        n.setRedirect(redirect());
+        if (isArray()) {
+            n.componentType = redirect().getComponentType();
+        }
+        return n;
+    }
+
     public ClassNode getPlainNodeReference() {
         if (ClassHelper.isPrimitiveType(this)) return this;
         ClassNode n = new ClassNode(name, modifiers, superClass, null, null);
```

Comment:
We could remove the duplicated code by invoking the method:
`return getPlainPrimitiveReference()`

Reply:
That would be possible. Let's see first if others feel the method makes sense. We could remove the new method and just always call `getPlainNodeReference()` if we removed the primitive check line. I just felt that might be more likely to break existing external code.